datalad_next/iter_collections/zipfile.py

```diff
@@ -75,7 +86,11 @@ def _get_zipfile_item(zip_info: zipfile.ZipInfo) -> ZipfileItem:
         else FileSystemItemType.file
     )
     return ZipfileItem(
-        name=PurePosixPath(zip_info.filename),
+        name=(
+            ZipFileDirPath(zip_info.filename)
+            if zip_info.is_dir()
```

Comment:
`is_dir()` is already called a few lines above, we already know whether this is a directory at this point.

Reply:
Fixed in db878f5d15765651ea56d8b8c6fc081478213f30